cmd/fluxsvc/main.go

```diff
@@ -184,27 +184,16 @@ func main() {
 	}
 
 	// Release workers.
-	waitForWorkers := make(chan struct{})
-	shutdownWorkers := make(chan struct{})
 	{
-		workers := sync.WaitGroup{}
-
-		// for each worker goroutine
-		{
-			workers.Add(1)
-			worker := release.NewWorker(rjs, instancer, releaseMetrics, logger)
-			go func() {
-				defer workers.Done()
-				worker.Work(shutdownWorkers)
-			}()
-		}
-
-		go func() {
-			workers.Wait()
-			close(waitForWorkers)
+		worker := jobs.NewWorker(rjs, instancer, releaseMetrics, logger)
+		defer func() {
+			if err := worker.Stop(shutdownTimeout); err != nil {
+				logger.Log("err", err)
+			}
 		}()
+		worker.Work()
```

Comment:
This will block, as things stand. Either the worker should start its own goroutines (which I think is what's intended), or this should be `go worker.Work()`.